halo2_proofs/src/plonk/circuit.rs

```diff
@@ -1357,6 +1573,38 @@ impl<F: Field> ConstraintSystem<F> {
         tmp
     }
 
+    /// Requests a challenge that is usable after the given phase.
+    pub fn challenge_usable_after(&mut self, phase: Phase) -> Challenge {
+        self.assert_no_phase_skipped(phase, "Challenge");
```

Comment:
If we're requiring that phases are assigned in order, then we should be checking here that `phase` exists, not `previous_phase` (since this challenge only makes sense if there are columns in `phase`).

Reply:
Ah I didn't think about that. The assertion helper to check if specified phase exists is refactored in [`6051345`](https://github.com/zcash/halo2/pull/593/commits/6051345b999acca2c5110dce01ce3cfa58e321b7#diff-9ee6d800f38a8bebc2692cd214db8ed87d3a800e9f42c15a4c53f6f6fac9877cR1592-R1622).